python/pyspark/serializers.py

```diff
@@ -378,6 +379,29 @@ class ArrowStreamPandasUDFSerializer(ArrowStreamPandasSerializer):
     Serializer used by Python worker to evaluate Pandas UDFs
     """
 
+    def __init__(self, timezone, safecheck, assign_cols_by_name, df_for_struct=False):
+        super(ArrowStreamPandasUDFSerializer, self) \
+            .__init__(timezone, safecheck, assign_cols_by_name)
+        self._df_for_struct = df_for_struct
+
+    def arrow_to_pandas(self, arrow_column, data_type):
+        from pyspark.sql.types import StructType, \
+            _arrow_column_to_pandas, _check_dataframe_localize_timestamps
+
+        if self._df_for_struct and type(data_type) == StructType:
+            import pandas as pd
+            import pyarrow as pa
+            column_arrays = zip(*[[chunk.field(i)
+                                   for i in range(chunk.type.num_children)]
+                                  for chunk in arrow_column.data.iterchunks()])
```

Comment:
it might be best to avoid dealing with array chunks and keep this high level if possible. Would it be possible to build the Pandas DataFrame by flattening the Arrow column, building a table from those and then converting that to pandas?  Something like this I think:

```
pdf = pa.Table.from_arrays(arrow_column.flatten()).to_pandas()
```
I'm not sure if the column names in the pdf would end up as expected though...

Reply:
`arrow_column.flatten()` is great! Then we can support pyarrow>=0.10.